src/cpuid/src/brand_string.rs

```diff
@@ -300,6 +293,21 @@ impl BrandString {
     }
 }
 
+trait NullTerminatorIndex {
+    fn null_terminator_index(&self) -> usize;
+}
+impl NullTerminatorIndex for &[u8] {
+    fn null_terminator_index(&self) -> usize {
+        let mut len = self.len();
+        {
+            while len > 0 && self[len - 1] == 0 {
```

Comment:
The major change here is moving the `len > 0` check before the `[]` accessor.

Ideally I'd do this as an .iter().rev().position() check but I assume there's a reason it's coded C style :)